flatdata-cpp/include/flatdata/Archive.h

```diff
@@ -150,8 +150,19 @@ Archive::describe_resource( std::ostream& stream,
                             bool too_large )
 {
     auto initialized = static_cast< bool >( resource );
+    std::string description;
+    if ( initialized )
+    {
+        description = std::is_base_of< Archive, ResourceType >::value
```

Comment:
What about:
* describing resourcing is sub-archives that failed to initialize? How does that look like now?
* describing resources in sub-archives that successfully loaded, how does that look like now?

I guess it might need to look like a tree-structure, e.g.:
```
resource1 : ...
archive_resource1: ...
    sub_resource1: ...
    sub_resource2: ...
```

Reply:
* Uninitialized is reported as an error only in mandatory case. But then it doesn't stop us from describing the `Archive` child resources instead of keeping it "N/A". Will adapt it.

* If everything is initialized, then then describe works but there is no indentation and everything is flattened out (it's flatdata after all). But I'll try to see if there's a nice way to get indentation.

Reply:
Would you suggest a tree structure with complete descriptions like so? :
![flatdata_tree_describe](https://user-images.githubusercontent.com/8684135/85199456-0db3be00-b2f0-11ea-8d90-5090befa8eb9.png)

Reply:
Looks a bit hacky... how about having a single combined list, but full resource names:

```
outermost
outer
outer.outer1
outer.outer2
outer.inner
outer.inner.inner
```

That could require passing down the path-prefix, and being able to disable headers/footers during recursive calls

Reply:
Well, first versions are mostly hacky. Anyhow, I've reworked it a bit based on your inputs. I would like to retain the tree structure and somehow include which ones are archives and which are plain structs. Here's how the latest version looks:
![describe_reworked](https://user-images.githubusercontent.com/8684135/85951335-cb7b2400-b962-11ea-8304-41f149c15076.png)

Let me know what you think.

Reply:
Looks good!